core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/json.kt

```diff
@@ -1077,4 +1086,141 @@ class JsonTests {
         val json = df.toJson()
         DataFrame.readJsonStr(json) shouldBe df
     }
+
+    @Test
+    fun `serialize images as base64 default`() {
+        testSerializeImagesAsBase64()
+    }
+
+    @Test
+    fun `serialize images as base64 gzip on resize off`() {
+        testSerializeImagesAsBase64(ImageEncodingOptions(encodeAsBase64 = true, options = ImageEncodingOptions.GZIP_ON))
+    }
+
+    @Test
+    fun `serialize images as base64 gzip off resize off`() {
+        testSerializeImagesAsBase64(ImageEncodingOptions(encodeAsBase64 = true, options = 0))
+    }
+
+    @Test
+    fun `serialize images as base64 gzip on resize 700`() {
+        testSerializeImagesAsBase64(
+            ImageEncodingOptions(
+                encodeAsBase64 = true,
+                imageSizeLimit = 700,
+                options = ImageEncodingOptions.GZIP_ON or ImageEncodingOptions.LIMIT_SIZE_ON
+            )
+        )
+    }
+
+    @Test
+    fun `serialize images with toString`() {
+        val images = readImagesFromResources()
+
+        val df = dataFrameOf(listOf("imgs"), images)
+        val jsonStr = df.toJsonWithMetadata(
+            20,
+            nestedRowLimit = 20,
+            imageEncodingOptions = ImageEncodingOptions(encodeAsBase64 = false)
+        )
+
+        val json = parseJsonStr(jsonStr)
+
+        for (i in images.indices) {
+            val row = (json[KOTLIN_DATAFRAME] as JsonArray<*>)[i] as JsonObject
+            val img = row["imgs"] as String
+
+            img shouldContain "BufferedImage"
+        }
+    }
+
+    private fun testSerializeImagesAsBase64(encodingOptions: ImageEncodingOptions? = null) {
```

Comment:
This test is quite abstract, somehow hard to tell what it tests given different options

Reply:
agree, I fixed that